nipype/interfaces/ants/preprocess.py

```diff
@@ -0,0 +1,358 @@
+        metric_str = ("-m {metric_type}[{fixed_image},{moving_image},"
+                      "{metric_weight},{radius_or_bins},{sampling_strategy},"
+                      "{sampling_percentage}]")
+        format_args = {}
+        metric_args = ["metric_type", "fixed_image", "moving_image",
+                       "metric_weight", "radius_or_bins", "sampling_strategy",
+                       "sampling_percentage"]
+        for metric_arg in metric_args:
+            if isdefined(getattr(self.inputs, metric_arg)):
+                format_args[metric_arg] = getattr(self.inputs, metric_arg)
+        return metric_str.format(**format_args)
+
+    def _format_transform(self):
+        transform_str = "-t {}[{}]"
+        if (isdefined(self.inputs.transformation_model)
+                and isdefined(self.inputs.gradient_step_length)):
+            return transform_str.format(self.inputs.transformation_model,
+                                        self.inputs.gradient_step_length)
+        raise ValueError("Unable to format transformation_model argument")
+
+    def _format_output(self):
+        if (isdefined(self.inputs.output_transform_prefix)
+                and isdefined(self.inputs.output_warped_image)
+                and isdefined(self.inputs.output_average_image)):
+            return "-o [{},{},{}]".format(
+                self.inputs.output_transform_prefix,
+                self.inputs.output_warped_image,
+                self.inputs.output_average_image
+            )
+        elif isdefined(self.inputs.output_average_image):
+            return "-o {}".format(self.inputs.output_average_image)
+        else:
+            raise ValueError("Unable to format output due to lack of inputs.")
+
+    def _list_outputs(self):
+        outputs = self._outputs().get()
+        if isdefined(self.inputs.output_average_image):
+            outputs['average_image'] = (
+                os.path.abspath(self.inputs.output_average_image)
+            )
+        if isdefined(self.inputs.output_warped_image):
+            outputs['warped_image'] = (
+                os.path.abspath(self.inputs.output_warped_image)
+            )
+        if isdefined(self.inputs.output_transform_prefix):
+            fname = '{}MOCOparams.csv'.format(
+                self.inputs.output_transform_prefix
+            )
+            outputs['composite_transform'] = os.path.abspath(fname)
+        if (isdefined(self.inputs.write_displacement) and
+                isdefined(self.inputs.output_transform_prefix) and
+                self.inputs.write_displacement is True):
+            fname = '{}Warp.nii.gz'.format(self.inputs.output_transform_prefix)
+            outputs['displacement_field'] = os.path.abspath(fname)
+        return outputs
+
+class Matrix2FSLParamsInputSpec(BaseInterfaceInputSpec):
+    matrix = File(
+        exists=True,
+        desc='Motion crrection matrices to be converted into FSL style motion parameters',
+        mandatory=True
+    )
```

Comment:
Sorry for suggesting API changes, but better before merge than after. Feel free to ignore this if I'm being too fiddly.

1. `Matrix2FSLParams` doesn't really imply "ANTs MC" to me. I know this is in `ants/preprocess.py`, but maybe indicate the source a little more clearly? `MotionCorr2FSLParams`?

2. There's a really handy nipype feature:

```Python
class Matrix2FSLParamsInputSpec:
    ants_matrix = File(exists=True, mandatory=True,
                       desc='Motion correction matrices to be converted into FSL-style motion parameters')
    fsl_params = File(name_template='%s.par', name_source='ants_matrix',
                      desc='FSL parameter file')

class Matrix2FSLParamsInputSpec:
    fsl_params = File(exists=True, desc='FSL parameter file')
```

This lets you entirely drop the `_list_outputs` function. (You'd need to make the entry in the output spec match the one in the input spec.)

3. Also, there's a typo in "correction". And I'd use "FSL-style" instead of "FSL style".